src/main/java/sirius/kernel/commons/Value.java

```diff
@@ -1298,6 +1300,46 @@ public LocalDateTime asLocalDateTimeOfEpochSeconds(LocalDateTime defaultValue) {
         return LocalDateTime.ofInstant(temporal, ZoneId.systemDefault());
     }
 
+    /**
+     * Parses a date using the given formatter. Tries parsing with the format for the given fallback language in case
```

Comment:
this method looks overly complex.

Maybe we add two versions:
public Value tryParseDate(DateTimeFormatter formatter) { ... } and
public Value tryParseDate(String languageCode) { return tryParseDate(NLS.getDateFormat(languageCode)); }

The first method always checks for null or if the wrapped value is already a local date and then just returns self.

This way, one could replicate the logic by calling value.tryParseDate(FORMAT).tryParseDate("en").tryParseDate(NLS.getCurrentLanguage()).asOptional().orElseThrow("Invalid Date Format")

Reply:
Great suggestion 👍  This simplifies the code here a lot.

With these changes, a parsing retry-chain now looks like this:
```
value.tryParseLocalDate(DATE_FORMATTER)
     .tryParseLocalDate("fr")
     .tryParseLocalDate(NLS.getCurrentLang())
     .tryParseLocalDate(DateTimeFormatter.ISO_LOCAL_DATE)
     .asOptional(LocalDate.class)
     .orElseThrow(() -> Exceptions.createHandled()
                                  .withNLSKey("invalidDateFormat")
                                  .set("expectedFormat", DATE_FORMAT)
                                  .handle());